src/lua/compat.lua

```diff
@@ -0,0 +1,307 @@
+-- compat.lua -- internal module intended to solve compatibility problems in
+-- different parts of Tarantool. Introduced in gh-7000, see also gh-6912.
+
+local NEW = true
+local OLD = false
+
+-- luacheck: no unused
+local internals = require('internal.compat')
+
+local options_format = {
+    default  = 'string',
+    brief    = 'string',
+    obsolete = 'string',
+}
+
+-- Contains options descriptions in following format:
+-- * default  (string)
+-- * brief    (string)
+-- * obsolete ('X.Y' version or nil)
+-- * current  (boolean, true for 'new')
+-- * selected (boolean)
+-- * action   (function)
+local options = { }
+
+-- Array with option names in order of addition.
+local options_order = { }
+
+local help = [[
+Tarantool compatibility module.
+To get help, see the Tarantool manual at https://github.com/tarantool/tarantool/wiki/compat .
+Available commands:
+
+    compat.help()                   -- show this help
+    compat.<option_name>            -- list option info
+    compat.<option_name> = 'old'    -- set desired value to option, could be 'old', 'new', 'default'
+    compat{<option_name> = 'new'}   -- set listed options to desired values ('old', 'new', 'default')
+    compat.dump()                   -- get Lua command that sets up different compat with same options as current
+    compat.add_option()             -- add new option by providing a table with name, default, brief, obsolete
+                                       and action function
+]]
+
+-- Returns table with all info on option.
+local function serialize_option(key)
+    assert(key)
+    local result = {
+        brief = options[key].brief,
+        default = options[key].default,
+        obsolete = options[key].obsolete,
+    }
+
+    if not options[key].selected then
+        result.current = 'default'
```

Comment:
In `serialize_option` we should not. It returns table like:
```yaml
---
 - current: default
   brief: <...>
   default: new
...
```

Reply:
But we do it in `serialize_compat()`. Why not here?

Reply:
Because, default value is right underneath. Overall I think, it is a good idea to allow something like `compat.foo.current == 'default'`.
If we have `'default (new)'` in current, the command wouldn't be so clean.

Reply:
Well, let's stick on the simpler way and leave `default`.